internal/dataplane/parser/parser.go

```diff
@@ -258,21 +258,90 @@ func findPort(svc *corev1.Service, wantPort kongstate.PortDef) (*corev1.ServiceP
 }
 
 func getUpstreams(
-	log logrus.FieldLogger, s store.Storer, serviceMap map[string]kongstate.Service) []kongstate.Upstream {
+	log logrus.FieldLogger,
+	s store.Storer,
+	serviceMap map[string]kongstate.Service,
+) []kongstate.Upstream {
 	upstreamDedup := make(map[string]struct{}, len(serviceMap))
 	var empty struct{}
 	upstreams := make([]kongstate.Upstream, 0, len(serviceMap))
 	for _, service := range serviceMap {
-		name := fmt.Sprintf("%s.%s.%s.svc", service.Backend.Name, service.Namespace, service.Backend.Port.CanonicalString())
+		if len(service.Backends) == 0 {
+			log.WithField("service_name", *service.Name).Warnf("skipping service - no backends configured")
+			continue
```

Comment:
After thinking it over, we can't reach this right now (at least with `HTTPRoutes`) because translation will fail on lack of backends. Removed this in 4c0914451442bc1211753cad1c3d1a013dbe73f3